webcompat/helpers.py

```diff
@@ -722,3 +722,25 @@ def ab_init(response):
             response.set_cookie(exp_id, var, max_age=max_age)
 
     return response
+
+
+def get_extra_labels(form):
+    """Extract extra_labels.
+
+    If extra_labels param exists in current session, use it,
+    otherwise use the value coming from form. Add form-v2-experiment
+    label if the experiment cookie exists.
+    """
+
+    extra_labels = session.pop('extra_labels', None)
+
+    if not extra_labels and 'extra_labels' in form and form['extra_labels']:
```

Comment:
@ksy36 This part of the code is slightly hard to read and I'm not sure it does what is expected.

```python
>>> form = {'a': 1, 'b': 2, 'c': 3}
>>> extra_labels = None
>>> not extra_labels and 'b' in form and form['b']
2
>>> not None
True
>>> 'b' in form
True
>>> form['b']
2
>>> not None and True and 2
2
```

Reply:
So for this: 

if `extra_labels` is not in the session, I check whether 'extra_labels' property exists in form and that it's not an empty string, otherwise if passing an empty string to `json.loads` I get this error `json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)`

basically this test case covers this:
`self.assertEqual(get_extra_labels({'extra_labels': ''}), None)`

Reply:
but yeah it's hard to read, maybe I'll split this into separate functions

Reply:
I've refactored this and added some more tests here https://github.com/webcompat/webcompat.com/pull/3012/commits/aabf44faf6285337e5a357103df47c249e42f5c5

Reply:
See https://github.com/webcompat/webcompat.com/pull/3012#commitcomment-35858847
and https://github.com/webcompat/webcompat.com/pull/3012#commitcomment-35858866